frontend/packages/operator-lifecycle-manager/src/components/deprecated-operator-warnings/DeprecatedOperatorWarnings.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { Label, FormAlert, Alert, Tooltip } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { YellowExclamationTriangleIcon } from '@console/dynamic-plugin-sdk/src/api/core-api';
+import { DeprecatedOperatorWarning } from '@console/operator-lifecycle-manager/src/types';
+
+export const DeprecatedOperatorWarningBadge: React.FC<DeprecatedOperatorWarningBadge> = ({
+  deprecation,
+  className,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <Label
+      color="orange"
+      className={className}
+      icon={
+        <Tooltip content={t('olm~Deprecated: {{message}}', { message: deprecation?.message })}>
+          <YellowExclamationTriangleIcon />
+        </Tooltip>
+      }
+    >
+      {t('olm~Deprecated')}
+    </Label>
+  );
+};
+
+export const DeprecatedOperatorWarningIcon: React.FC<DeprecatedOperatorWarning> = ({
+  deprecation,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <Tooltip content={t('olm~Deprecated: {{message}}', { message: deprecation?.message })}>
```

Comment:
Since `message` is optional, you should make the `<Tooltip>` optional.

Reply:
Good point! But I just double check the API docs, the `message` is required. Will make the change in the type.